core/thread.go

```diff
@@ -543,42 +538,35 @@ func (t *Thread) indexBlock(commit *commitResult, blockType pb.Block_BlockType,
 	return nil
 }
 
-// handleHead determines whether or not a thread can be fast-forwarded or if a merge block is needed
-// - parents are the parents of the incoming chain
-func (t *Thread) handleHead(inbound mh.Multihash, parents []string) (mh.Multihash, error) {
-	head, err := t.Head()
+// handleHead determines what the next set of HEADs will be.
+// One of three situations will occur:
+// 1) fast-forward: the inbound parents are identical to current heads (heads -> [inbound])
+// 2) partial-split: at least one inbound parent matches a current head (heads -> [inbound, unmatched...])
+// 3) full-split: zero inbound parents match a current head (heads -> [inbound, current...])
+func (t *Thread) handleHead(inbound string, inboundParents []string) error {
```

Comment:
@carsonfarmer will need your eyes to double check me here. This feels much better... no more merge blocks, just maintain multiple HEADs until they can be tied back together with a local update (or a magical remote update (situation 1).

Reply:
I also noticed it is possible to have a 'pending' parent. Is this just the default until a remote or local update comes in?